utils/java/mvn.go

```diff
@@ -231,5 +228,9 @@ func (mdt *MavenDepTreeManager) CreateTempDirWithSettingsXmlIfNeeded() (tempDirP
 	if mdt.depsRepo != "" {
 		err = mdt.createSettingsXmlWithConfiguredArtifactory(tempDirPath)
 	}
+	if err != nil {
+		err = errors.Join(err, clearMavenDepTreeRun())
+		clearMavenDepTreeRun = nil
+	}
```

Comment:
Now that the clean, in the case of error, occurs inside the `CreateTempDirWithSettingsXmlIfNeeded`
you need to also take care of the code that uses it...
`RunMavenDepTree:86` needs to be removed.
also I can see a call for `RunMavenDepTree` in case of error at: `buildMavenDependencyTree:66`
Make sure to take care of all the cases for this